balrogscript/script.py

```diff
@@ -1,70 +1,27 @@
 #!/usr/bin/env python
+"""Balrog script
+"""
 from copy import deepcopy
 import json
-import jsonschema
 import logging
 import os
-import re
 import sys
 
+from balrogscript.task import (validate_task_schema, get_task,
+                               get_upstream_artifacts, get_manifest,
+                               get_task_server)
 
-log = logging.getLogger(__name__)
-
-
-def get_manifest(config):
-    # assumes a single upstreamArtifact and single path
-    task_id = config['upstream_artifacts'][0]['taskId']
-    path = os.path.join(config['work_dir'], "cot",
-                        task_id, config['upstream_artifacts'][0]['paths'][0])
-    log.info("Reading manifest file %s" % path)
-    try:
-        with open(path, "r") as fh:
-            manifest = json.load(fh)
-    except (ValueError, OSError) as e:
-        log.critical("Can't load manifest from {}!\n{}".format(path, e))
-        sys.exit(3)
-    return manifest
-
-
-def verify_task_schema(config, task_definition):
-    schema_path = os.path.join(
-        os.path.dirname(os.path.dirname(__file__)), config['schema_file']
-    )
-    with open(schema_path) as fh:
-        schema = json.load(fh)
-
-    try:
-        jsonschema.validate(task_definition, schema)
-    except jsonschema.exceptions.ValidationError as exc:
-        log.critical("Can't validate schema!\n{}".format(exc))
-        sys.exit(3)
-
-
-def load_task(config):
-    task_file = os.path.join(config['work_dir'], "task.json")
-    with open(task_file, 'r') as f:
-        task_definition = json.load(f)
-
-    verify_task_schema(config, task_definition)
-    upstream_artifacts = task_definition['payload']['upstreamArtifacts']
-    for scope in task_definition['scopes']:
-        if scope.startswith("project:releng:balrog:"):
-            signing_cert_name = scope.split(':')[-1]
-            if re.search('^[0-9A-Za-z_-]+$', signing_cert_name) is not None:
-                break
-            log.warning('scope {} is malformed, skipping!'.format(scope))
-    else:
-        log.critical("no balrog scopes!")
-        sys.exit(3)
 
-    return upstream_artifacts
+log = logging.getLogger(__name__)
 
 
 def create_submitter(e, balrog_auth, config):
     from balrog.submitter.cli import NightlySubmitterV4, ReleaseSubmitterV4  # noqa: E402
     auth = balrog_auth
 
     if "previousVersion" in e and "previousBuildNumber" in e:
+        # TODO: to rewrite this only for CMAR. partials are handled externally
+        # by funsize
```

Comment:
At some point, funsize will generate the partial mars, but balrogscript will submit them.